examples/ensemble/plot_stack_predictors.py

```diff
@@ -0,0 +1,115 @@
+"""
+=================================
+Combine predictors using stacking
+=================================
+
+Stacking refers to a method to blend estimators. In this strategy, some
+estimators are individually fitted on some training data while a final
+estimator is trained using the stacked predictions of these base estimators.
+
+In this example, we illustrate the use case in which different regressors are
+stacked together and a final linear penalized regressor is used to output the
+prediction. We compare the performance of each individual regressor with the
+stacking strategy. Stacking slightly improves the overall performance.
+
+"""
+print(__doc__)
+
+# Authors: Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# License: BSD 3 clause
+
+###############################################################################
+# The function ``plot_regression_results`` is used to plot the predicted and
+# true targets.
+
+import matplotlib.pyplot as plt
+
+
+def plot_regression_results(ax, y_true, y_pred, title, scores):
+    """Scatter plot of the predicted vs true targets."""
+    ax.plot([y_true.min(), y_true.max()],
+            [y_true.min(), y_true.max()],
+            '--r', linewidth=2)
+    ax.scatter(y_true, y_pred, alpha=0.2)
+
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    ax.get_xaxis().tick_bottom()
+    ax.get_yaxis().tick_left()
+    ax.spines['left'].set_position(('outward', 10))
+    ax.spines['bottom'].set_position(('outward', 10))
+    ax.set_xlim([y_true.min(), y_true.max()])
+    ax.set_ylim([y_true.min(), y_true.max()])
+    ax.set_xlabel('Measured')
+    ax.set_ylabel('Predicted')
+    extra = plt.Rectangle((0, 0), 0, 0, fc="w", fill=False,
+                          edgecolor='none', linewidth=0)
+    ax.legend([extra], [scores], loc='upper left')
+    ax.set_title(title)
+
+
+###############################################################################
+# Stack of predictors on a single data set
+###############################################################################
+# It is sometimes tedious to find the model which will best perform on a given
```

Comment:
And somehow, I think the logic here is strange. I'm still not sure whether it's good to throw "bad" estimators to a stacker. It's true that a stacker can somehow filter "bad" estimators (e.g., by giving lower weight to them), but I think these "bad" estimators will still influence the performance of a stacker. 
I think one should use cross validation to check the performance of all the estimators and throw some "good" estimators to a stacker to produce a "better" estimator.
But I'm not sure and I think this example is acceptable.

Reply:
I agree. Stacking is mostly useful to slightly improve the accuracy by combining the strength of good models (hopefully diverse enough to get not too correlated errors).

Throwing in bad models is likely to negatively impact the performance of the ensemble (and it's computationally wasteful).

Reply:
OK, so I replaced `AdaBoostRegressor` and `KNNRegressor` by `HistGradientBoostingRegressor` and `RandomForestRegressor` to have only strong learner and changed the conclusion slightly. I will add a bit of timing as well to show that training a stack learner is computationally expensive